derive/src/traits.rs

```diff
@@ -202,6 +205,24 @@ impl Derivable for Zeroable {
   fn explicit_bounds_attribute_name() -> Option<&'static str> {
     Some("zeroable")
   }
+
+  fn perfect_derive_fields(input: &DeriveInput) -> Option<Fields> {
+    match &input.data {
+      Data::Struct(struct_) => Some(struct_.fields.clone()),
+      Data::Enum(DataEnum { variants, .. }) => {
+        let iter = VariantDiscriminantIterator::new(variants.iter());
+        for res in iter {
+          match res {
+            Ok((0, variant)) => return Some(variant.fields.clone()),
+            Ok(_) => (),
+            Err(_) => return None,
+          }
+        }
+        None
```

Comment:
Third occurrence of this pattern; it may warrant a helper function 🤔

Reply:
Latest push adds a helper function `get_zero_variant(enum: &DataEnum) -> Option<&Variant>`